app/models/annotation/AnnotationMutexService.scala

```diff
@@ -0,0 +1,119 @@
+package models.annotation
+
+import akka.actor.ActorSystem
+import com.scalableminds.util.accesscontext.GlobalAccessContext
+import com.scalableminds.util.time.Instant
+import com.scalableminds.util.tools.Fox
+import com.scalableminds.webknossos.datastore.helpers.IntervalScheduler
+import com.scalableminds.webknossos.schema.Tables.AnnotationMutexesRow
+import com.typesafe.scalalogging.LazyLogging
+import models.user.{UserDAO, UserService}
+import net.liftweb.common.Full
+import play.api.inject.ApplicationLifecycle
+import play.api.libs.json.{JsObject, Json}
+import utils.{ObjectId, WkConf}
+import utils.sql.{SimpleSQLDAO, SqlClient}
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+case class AnnotationMutex(annotationId: ObjectId, userId: ObjectId, expiry: Instant)
+
+case class MutexResult(canEdit: Boolean, blockedByUser: Option[ObjectId])
+
+class AnnotationMutexService @Inject()(val lifecycle: ApplicationLifecycle,
+                                       val system: ActorSystem,
+                                       wkConf: WkConf,
+                                       userDAO: UserDAO,
+                                       userService: UserService,
+                                       annotationMutexDAO: AnnotationMutexDAO)
+    extends IntervalScheduler
+    with LazyLogging {
+
+  override protected def tickerInterval: FiniteDuration = 1 hour
+
+  override protected def tick(): Unit = {
+    logger.info("Cleaning up expired annotation mutexes...")
+    annotationMutexDAO.deleteExpired()
+    ()
+  }
+
+  private val defaultExpiryTime = wkConf.WebKnossos.Annotation.Mutex.expiryTime
+
+  def tryAcquiringAnnotationMutex(annotationId: ObjectId, userId: ObjectId)(
+      implicit ec: ExecutionContext): Fox[MutexResult] =
+    this.synchronized {
```

Comment:
I was assuming that the mutex is handled via an sql transaction instead of local locking, but probably also fine :+1:

Reply:
Yes, that would be the next iteration of this feature. Because of this, I added the noticeSlowRequest with a tight timeout, so that we will know when this variant blocks too many requests. My guess is that it is not a problem, and we can upgrade to an sql-only version later. It would probably be a bit harder to write and understand, as it would need to implement more of the logic in sql, be a full test-and-set, depending on whether the locking user is the requesting one etc.